specs/overview.md

```diff
@@ -0,0 +1,170 @@
+
+This document is a high-level technical overview of the Optimism protocol. It aims to explain how the protocol works in an informal manner, and direct readers to other parts of the specification so that they may learn more. A basic understanding of how optimistic rollups work is a prerequisite to reading this document.
+
+## Components
+
+![Components](./assets/components.svg)
+
+### L1 Components
+
+**DepositFeed**
+
+- A smart contract that processes deposits from L1 to L2.
+- Though the name `DepositFeed` implies depositing tokens, in reality any valid Ethereum transaction can be deposited into the `DepositFeed`.
+
+**BatchInbox**
+
+- An address that the Batch Submitter uses to submit transaction batches.
+- Counter-intuitively, this address does not contain a smart contract since we can instead send calldata directly to an EOA.
+
+**L2OutputOracle**
+
+- A smart contract that stores L2 output roots.
+
+### L2 Components
+
+**Rollup Node**
+
+- A standalone binary that sits between the execution engine and L1.
+- Derives data from L1, and handles L1 reorgs.
+- Validates incoming block data.
+- Prepares execution payloads as per the [Engine API](https://hackmd.io/@n0ble/consensus_api_design_space).
+- Distributes unsubmitted blocks to other rollup nodes.
+
+**Execution Engine (EE)**
+
+- A vanilla Geth node with minor modifications to support Optimism.
+- Syncs state to other nodes on the Optimism network via a P2P network.
+- Maintains L2 state.
+- Serves the Engine API to the Rollup Node.
+
+**Batch Submitter**
+
+- A background process that submits transaction batches to the `BatchInbox` address.
+
+**Output Submitter**
+
+- A background process that submits L2 state roots to the `L2OutputOracle`.
+
+### Types of Rollup Nodes
+
+There are two types of nodes on Optimism: sequencers, and verifiers.
+
+- **Sequencers** are responsible for accepting user-sent and deposit transactions, ordering them into L2 blocks, and submitting batch/output roots to L1.
+- **Verifiers** reconstruct the L2 chain from data on l1.
+
+Both node types run the EE and Rollup Node to maintain their view of the rollup state.
+
+Sequencers, however, also perform block production and batch/output submission. There is currently only one sequencer. We may support distributed sequencing in the future.
+
+### Transaction/Block Propagation
+
+**Spec links:**
+
+- [Execution Engine](specs/exec-engine.md)
+
+Since the EE uses Geth under the hood, Optimism uses Geth's built-in peer-to-peer network and transaction pool to propagate transactions. The same network can also be used to propagate submitted blocks and support snap-sync. 
+
+Unsubmitted blocks, however, are propagated using a separate peer-to-peer network of Rollup Nodes. This is optional, however, and is provided as a convnience to lower latency for users.
+
+The below diagram illustrates how the sequencer and verifiers fit together:
+
+![Propagation](./assets/propagation.svg)
```

Comment:
I like having the unsafe block propagation arrow be bidirectional because it implies the existence of a peer-to-peer network that gossips data rather than a client-server network where verifiers always read from the sequencer.